### PR TITLE
Improve title API

### PR DESCRIPTION
**Pull request by Techcable**
Uses title objects instead of individual update methods.

I haven't created a test plugin to test this, just tell me if you want me to do that.
It should be somewhat stable, since its based on the already tested sportbukkit api.
I have verified this doesn't crash the server though.

### CODE REVIEW

File: Spigot-Server-Patches/0094-Player-Tab-List-and-Title-APIs.patch [hunk 1]
```diff
@@ -159,6 +176,24 @@ index dba0520..cabef35 100644
 +    }
 +
 +    @Override
++    public void sendTitle(Title title) {
++        Preconditions.checkNotNull(title, "Title is null");
++        setTitleTimes(title.getFadeIn(), title.getStay(), title.getFadeOut());
++        setSubtitle(title.getSubtitle() == null ? new BaseComponent[0] : title.getSubtitle());
```
Comment (kashike) [inline Spigot-Server-Patches/0094-Player-Tab-List-and-Title-APIs.patch:182]: Why construct a `new BaseComponent[0]`?

Comment (Techcable) [inline Spigot-Server-Patches/0094-Player-Tab-List-and-Title-APIs.patch:182]: Does setSubtitle() accept a null to clear the existing subtitle?

Comment (thekeenant) [inline Spigot-Server-Patches/0094-Player-Tab-List-and-Title-APIs.patch:182]: I don't believe it does, throws an NPE to the client?

**Comment by zachbr** [comment]
status?

**Comment by Techcable** [comment]
Finished and ready to pull

**Pull request merged**